### DISCUSSION TITLE
Filter per byte value

### DISCUSSION THREAD

**Discussion by rcavedon** [TOSUN-Shanghai/TSMaster#74] [Q&A]
Hi,
I have a question about the filtering of messages.
For me, it's clear how to filter a canbus message for ID but not if I want to filter for a specific byte in the data frame.
I guess a Script C is needed but I'm not able to understand what to write inside the script.

Thank you in advance,
  Riccardo

**Comment by freshhope** [comment]
Hi Riccardo,

It is possible in many ways, you can see one of the following solutions in the following screenshot:
![image](https://user-images.githubusercontent.com/16358754/200732003-88cfa5a5-2218-4995-8f4b-145001a0963f.png)

**Reply by rcavedon** [reply]
Thank you @freshhope for the suggestion but I'm not able to filter correctly yet.
I've applied the code to a C script with the following configuration
![image](https://user-images.githubusercontent.com/26982041/200782701-4705be48-5f85-440e-9b96-defc68f51351.png)
I tried to filter in order to see only a message with byte 3 == 0x23 but I can see in the trace also the other messages. Is there a wait to discharge the messages?
![image](https://user-images.githubusercontent.com/26982041/200783991-a4fcffc3-76d2-4131-85a8-f229709ff11a.png)
![image](https://user-images.githubusercontent.com/26982041/200784531-d2fe17ae-12ab-473f-bead-8f82ffac838e.png)

Thank you,
  Riccardo

**Comment by freshhope** [comment]
Hi Riccardo,

Your filter logic is a dynamic logic as it checks for data bytes, all current TSMaster filters are static, so you need to do the following:

[1] add a new virtual channel for filtered message display, for example, channel 2
[2] before injecting the message, direct the message's channel to that virtual channel: 
    a.FIdxChn = CH2;
    com.inject_can_message((PCANFD)&a);
[3] you can see the injected message in trace, and you can also set trace filter to only display channel 2 messages
[4] then the trace is showing messages only from your filter
